### PR TITLE
Revert sudo user

### PR DESCRIPTION
**Pull request by runcom**
Conversation here: https://github.com/docker/docker/pull/20763#discussion_r54355457
Basically that was too confusing
- add a note about docker with sudo in docs
- move ".docker" string to a const

### CODE REVIEW

**Comment by runcom** [comment]
ping @duglin @thaJeztah @vdemeester

**Comment by duglin** [comment]
Weird that the tests didn't run.

**Comment by thaJeztah** [comment]
@duglin I also had huge delays triggering builds from IRC, so may be that GitHub has issues

**Comment by thaJeztah** [comment]
triggered CI from IRC

**Comment by thaJeztah** [comment]
change LGTM, although I wasn't against the original change, but I get why it may not be the best thing. It's just a thing we're a bit stuck with, due to sudo being needed (or not), depending on the situation (local/remote daemon etc).

File: docs/reference/commandline/cli.md [hunk 1]
```diff
@@ -78,8 +78,9 @@ For example:
 Instructs Docker to use the configuration files in your `~/testconfigs/`
 directory when running the `ps` command.
 
-> **Note**: If you run docker commands with `sudo`, Docker first looks for a configuration
-> file in `/root/.docker/`, before looking in `~/.docker/` for the user that did the sudo call.
+> **Note**: If you run docker commands with `sudo`, Docker looks for a configuration
+> file in `/root/.docker/` by default. In order to make it look for a configuration
+> file for the user who did the call, you have to use `sudo -u <username>`.
```
Comment (phemmer) [inline docs/reference/commandline/cli.md]: This isn't accurate. If I'm reading the code right, it's going to look for a configuration file in `$HOME/.docker`. This doesn't necessarily mean `/root` if `sudo` is used. `sudo` has a `-E` option to preserve environment variables, including `$HOME`.

Comment (runcom) [inline docs/reference/commandline/cli.md]: Right, I'll rephrase it to mention by default HOME with sudo will resolve to the root home, thx

**Comment by runcom** [comment]
> Weird that the tests didn't run.

I think because I hadn't signed the revert commit

**Comment by duglin** [comment]
LGTM once janky is happy

**Comment by vdemeester** [comment]
LGTM :sweat_smile:

**Comment by thaJeztah** [comment]
Reverted in https://github.com/docker/docker/pull/20780, so removing impact labels
